client/src/leap/soledad/client/http_target.py

```diff
@@ -0,0 +1,602 @@
+        Insert a received document into the local replica.
+
+        :param idx: The index count of the current operation.
+        :type idx: int
+        :param total: The total number of operations.
+        :type total: int
+        :param response: The body and headers of the response.
+        :type response: tuple(str, dict)
+        """
+        new_generation, new_transaction_id, number_of_changes, doc_id, \
+            rev, content, gen, trans_id = \
+            self._parse_received_doc_response(response)
+        if doc_id is not None:
+            # decrypt incoming document and insert into local database
+            # -------------------------------------------------------------
+            # symmetric decryption of document's contents
+            # -------------------------------------------------------------
+            # If arriving content was symmetrically encrypted, we decrypt it.
+            # We do it inline if defer_decryption flag is False or no sync_db
+            # was defined, otherwise we defer it writing it to the received
+            # docs table.
+            doc = SoledadDocument(doc_id, rev, content)
+            if is_symmetrically_encrypted(doc):
+                if self._queue_for_decrypt:
+                    self._sync_decr_pool.insert_encrypted_received_doc(
+                        doc.doc_id, doc.rev, doc.content, gen, trans_id,
+                        idx)
+                else:
+                    # defer_decryption is False or no-sync-db fallback
+                    doc.set_json(decrypt_doc(self._crypto, doc))
+                    self._insert_doc_cb(doc, gen, trans_id)
+            else:
+                # not symmetrically encrypted doc, insert it directly
+                # or save it in the decrypted stage.
+                if self._queue_for_decrypt:
+                    self._sync_decr_pool.insert_received_doc(
+                        doc.doc_id, doc.rev, doc.content, gen, trans_id,
+                        idx)
+                else:
+                    self._insert_doc_cb(doc, gen, trans_id)
+            # -------------------------------------------------------------
+            # end of symmetric decryption
+            # -------------------------------------------------------------
+        msg = "%d/%d" % (idx, total)
+        signal(SOLEDAD_SYNC_RECEIVE_STATUS, msg)
+        logger.debug("Soledad sync receive status: %s" % msg)
+        return number_of_changes, new_generation, new_transaction_id
+
+    def _parse_received_doc_response(self, response):
+        """
+        Parse the response from the server containing the received document.
+
+        :param response: The body and headers of the response.
+        :type response: tuple(str, dict)
+
+        :return: (new_gen, new_trans_id, number_of_changes, doc_id, rev,
+                 content, gen, trans_id)
+        :rtype: tuple
+        """
+        # decode incoming stream
+        parts = response.splitlines()
```

Comment:
for some reason, I'm getting ints as the passed response
```
Traceback (most recent call last):
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 1105, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/python/failure.py", line 389, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/kali/leap/soledad/client/src/leap/soledad/client/sync.py", line 139, in sync
    defer_decryption=defer_decryption)
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 1105, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/python/failure.py", line 389, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/kali/leap/soledad/client/src/leap/soledad/client/http_target.py", line 270, in sync_exchange
    defer_decryption=defer_decryption)
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 1105, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/python/failure.py", line 389, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/kali/leap/soledad/client/src/leap/soledad/client/http_target.py", line 401, in _receive_docs
    number_of_changes, ngen, ntrans = yield d
  File "/home/kali/Virtualenvs/leap-bitmask/local/lib/python2.7/site-packages/Twisted-15.1.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 578, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/home/kali/leap/soledad/client/src/leap/soledad/client/http_target.py", line 507, in _insert_received_doc
    self._parse_received_doc_response(response)
  File "/home/kali/leap/soledad/client/src/leap/soledad/client/http_target.py", line 556, in _parse_received_doc_response
    parts = response.splitlines()
AttributeError: 'int' object has no attribute 'splitlines'
```

also, it doesn't make sense the stated type of response, everything points to it being a str, isn't it?